modules/standard/IO.chpl

```diff
@@ -2984,6 +2990,17 @@ private inline proc _read_one_internal(_channel_internal:qio_channel_ptr_t,
                                        ref x:?t,
                                        loc:locale):syserr where _isIoPrimitiveTypeOrNewline(t) {
   var e:syserr = ENOERR;
+
+  var reader = new channel(writing=false, iokind.dynamic, locking=false,
+                           home=here,
+                           _channel_internal=_channel_internal,
+                           _readWriteThisFromLocale=loc);
+  var inJson = false;
+  var st = reader.styleElement(QIO_STYLE_ELEMENT_AGGREGATE);
+
+  if st == QIO_AGGREGATE_FORMAT_JSON then inJson = true;
+  reader._channel_internal = QIO_CHANNEL_PTR_NULL;
```

Comment:
Two things:

1. I can't understand why we'd create a `new channel` at this point. It seems to me we just need to check the styleElement of the current channel. When I pointed you to the example earlier, it didn't occur to me that we wouldn't have the `record channel` available. We can call the function to get the style element on the _channel_internal, with `var st = qio_channel_style_element(_channel_internal, QIO_STYLE_ELEMENT_AGGREGATE);`

2. I don't see a good reason to get the style element here in _read_one_internal/_write_one_internal and pass it to _read_text_internal/_write_text_internal. I think it should just make the query when handling an enum in  _read_text_internal/_write_text_internal.

Reply:
OK, I didn't know if there is such a function, earlier in code, as you mentioned before, there was a writer writer created, in some overload of _write_one_internal, but if there is such a function exists then its much better, thanks, its better to just get the st without creating another object.